e2e/common.go

```diff
@@ -1,12 +1,116 @@
 	ginkgo.GinkgoWriter.Write([]byte(fmt.Sprintf(msg, args...)))
 }
+
+// GetSamplePVC returns a sample pvc.
+func GetSamplePVC(storageClass string, quantity string, name string, volumemode k8sv1.PersistentVolumeMode) *k8sv1.PersistentVolumeClaim {
+	_, err := resource.ParseQuantity(quantity)
+	gomega.Expect(err).To(gomega.BeNil())
+
+	pvc := &k8sv1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: TestNamespace,
+		},
+		Spec: k8sv1.PersistentVolumeClaimSpec{
+			StorageClassName: &storageClass,
+			AccessModes:      []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteOnce},
+			VolumeMode:       &volumemode,
+			Resources: k8sv1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceStorage: resource.MustParse(quantity),
+				},
+			},
+		},
+	}
+	return pvc
+}
+
+// GetSamplePod returns a sample pod.
+func GetSamplePod(name string, pvcName string) *k8sv1.Pod {
+	pod := &k8sv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: TestNamespace,
+		},
+		Spec: k8sv1.PodSpec{
+			Volumes: []k8sv1.Volume{
+				{
+					Name: "storage",
+					VolumeSource: k8sv1.VolumeSource{
+						PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+						},
+					},
+				},
+			},
+			Containers: []k8sv1.Container{
+				{
+					Name:  "container",
+					Image: "public.ecr.aws/docker/library/nginx:latest",
+					Ports: []k8sv1.ContainerPort{
+						{
+							Name:          "http-server",
+							ContainerPort: 80,
+						},
+					},
+				},
+			},
+		},
+	}
+	return pod
+}
+
+// WaitForPVCBound waits for a pvc with a given name and namespace to reach BOUND phase.
+func WaitForPVCBound(pvc *k8sv1.PersistentVolumeClaim, namespace string, pod *k8sv1.Pod) error {
+	pvc, err := DeployManagerObj.GetK8sClient().CoreV1().PersistentVolumeClaims(namespace).Create(context.TODO(), pvc, metav1.CreateOptions{})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	_, err = DeployManagerObj.GetK8sClient().CoreV1().Pods(namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	lastReason := ""
+	timeout := 100 * time.Second
+	interval := 1 * time.Second
+
+	// Wait for namespace to terminate
+	err = utilwait.PollImmediate(interval, timeout, func() (done bool, err error) {
+		pvc, err = DeployManagerObj.GetK8sClient().CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.TODO(), pvc.Name, metav1.GetOptions{})
+		if err != nil {
+			lastReason = fmt.Sprintf("error talking to k8s apiserver: %v", err)
+			return false, nil
+		}
+		if pvc.Status.Phase != k8sv1.ClaimBound {
+			lastReason = fmt.Sprintf("waiting on pvc %s/%s to reach bound state, currently %s", pvc.Namespace, pvc.Name, pvc.Status.Phase)
+			return false, nil
+		}
+
+		return true, nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("%v: %s", err, lastReason)
+	}
+
```

Comment:
You also need to check pod is in a running the state?

Reply:
Added the function for same.